lib/federation.js

```diff
@@ -250,10 +252,25 @@ function buildFederationSchema (schema, isGateway) {
   })
 
   // Add all extensions
-  federationSchema = extendSchema(federationSchema, {
+  const extensionsDocument = {
     kind: Kind.DOCUMENT,
     definitions: extensions
-  })
+  }
+
+  // instead of relying on extendSchema internal validations
+  // we run validations in our code so that we can use slightly different rules
+  // as extendSchema internal rules are meant for regular usage
+  // and federated schemas have different constraints
+  const errors = validateSDL(extensionsDocument, federationSchema, compositionRules)
+  if (errors.length === 1) {
```

Comment:
I think this error handling is messy.

I would suggest always throwing an error with the message `Invalid Federation Schema` or similar, and always setting the errors property.

Reply:
I copied the approach from https://github.com/mcollina/fastify-gql/blob/master/index.js#L158, but I'm happy to simplify it if we don't mind taking a different approach.

(or I guess we can change that code to have the same behaviour?)

Reply:
What we have found in several other issues is that devs will not look into the `errors` property and open issues about invalid schema :/. See https://github.com/mcollina/fastify-gql/issues/215 and https://github.com/mcollina/fastify-gql/pull/216. I agree this is far from ideal but it provides a better developer experience.